src/java.base/share/classes/sun/security/util/math/intpoly/IntegerPolynomial.java

```diff
@@ -553,16 +551,22 @@ public MutableElement mutable() {
             return new MutableElement(limbs.clone(), numAdds);
         }
 
-        protected boolean isSummand() {
-            return numAdds < maxAdds;
-        }
-
         @Override
         public ImmutableElement add(IntegerModuloP genB) {
             assert IntegerPolynomial.this == genB.getField();
-            Element b = (Element) genB;
-            if (!(isSummand() && b.isSummand())) {
-                throw new ArithmeticException("Not a valid summand");
+            Element b = (Element)genB;
+
+            // Reduce if required.
+            // if (numAdds >= maxAdds) {
+            if (numAdds > 32 - bitsPerLimb) {
```

Comment:
if we allow this number of additions, `setProduct(SmallValue)` might overflow in the future. Currently it's safe - we only create a limited set of SmallValues, and they are all small enough to avoid this risk.
`getSmallValue` allows numbers up to `bitsPerLimb` bits. I think we can adjust that.

Reply:
Sorry, I did not get the ideas.  Did you meant to adjust the implementation of getSmallValue to allow 32 bits int value?

Reply:
Apologies, I was looking at the wrong branch, didn't notice `setProduct(SmallValue)` also does a reduce if required.